testing_app/test_driver/state_mgmt.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_driver/driver_extension.dart';
+import 'package:provider/provider.dart';
+import 'package:testing_app/models/favorites.dart';
+import 'package:testing_app/screens/favorites.dart';
+
+void main() {
+  // Create a reference to the state
+  // so that we can perform operations on it manually.
+  Favorites favoritesList;
+
+  // This line enables the extension.
+  enableFlutterDriverExtension(
+      // This handler gets data from the test file
+      // and then we use it to perform operations on the state object.
+      handler: (command) {
+    if (command == 'ADD') {
+      favoritesList.add(30);
+    } else if (command == 'REMOVE') {
```

Comment:
I've used this method for communication between the application and the test file. It's working fine. Please let me know if there is any other way that I should be using.